### PR TITLE
fix: Go enum schema changes

### PR DESCRIPTION
**Pull request by kridai**
# Fixes #

Updating Go code generation to support enum schema changes done as part of the pr https://github.com/twilio/twilio-oai/pull/71/files

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [ ] Run `make test-docker`
- [ ] Verify affected language:
    - [ ] Generate [twilio-go](https://github.com/twilio/twilio-go) from our [OpenAPI specification](https://github.com/twilio/twilio-oai) using the [build_twilio_go.py](./examples/build_twilio_go.py) using `python examples/build_twilio_go.py path/to/twilio-oai/spec/yaml path/to/twilio-go` and inspect the diff
    - [ ] Run `make test` in `twilio-go`
    - [ ] Create a pull request in `twilio-go`
    - [ ] Provide a link below to the pull request
- [ ] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [ ] I have read the [Contribution Guidelines](https://github.com/twilio/twilio-oai-generator/blob/main/CONTRIBUTING.md) and my PR follows them
- [ ] I have titled the PR appropriately
- [ ] I have updated my branch with the main branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added the necessary documentation about the functionality in the appropriate .md file
- [ ] I have added inline documentation to the code I modified

If you have questions, please create a GitHub Issue in this repository.

### CODE REVIEW

File: examples/twilio_api_v2010.yaml [hunk 1]
```diff
@@ -1,5 +1,30 @@
...
+      enum:
+        - DialVerb
+        - Trunking
+      type: string
+    payments_enum_bank_account_type:
+      enum:
+        - consumer-checking
+        - consumer-savings
+        - commercial-checking
+      type: string
```
Comment (childish-sambino) [inline examples/twilio_api_v2010.yaml:27]: I say just reduce these down to 1 enum to make the fixture more concise.

**Review by childish-sambino — COMMENTED** [review]
Tests failures need to be addressed.

File: examples/twilio_api_v2010.yaml (hunk 1 above)
Comment (charan678) [reply examples/twilio_api_v2010.yaml:27]: @childish-sambino Test are passing for go-lang. We did raise PR for java. As those changes in different commit. If you approve it then we can raise another PR for java

Comment (charan678) [reply examples/twilio_api_v2010.yaml:27]: We will remove extra enums before merging to main

Comment (childish-sambino) [reply examples/twilio_api_v2010.yaml:27]: Prefer to not make extra changes after getting approval. Think it's best to trim them down first, then review, then merge for a smooth process.

**Pull request closed without merging**